scripts/run_project_tests.py

```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/env python3
+import subprocess
+import argparse
+import sys
+import os
+from typing import Dict, List
+
+import yaml
+from colorama import Fore, init
+
+init(autoreset=True)
+
+"""
+A script to run unity and pytest tests in a host environment for Phoenix-RTOS.
+
+Example of config file:
+
+    campaign_name:
+        unity:
+            - binary1 args
+            - binary2
+        pytest:
+            - path1 args
+            - path2
+
+NEED TO KNOW:
+
+    Need vm.map_rnd_bits set to 28
+    You can use: sudo sysctl vm.mmap_rnd_bits=28
+    due to the issue reported here: https://github.com/phoenix-rtos/phoenix-rtos-project/issues/1032
+"""
+
+
+class TestSetupError(Exception):
+    pass
+
+
+def print_fancy(message: str, color: str, flush: bool = False, end: str = "\n") -> None:
+    """Prints a message with the specified color."""
+    print(f"{color}{message}", flush=flush, end=end)
+
+
+def run_test(test_cmd: List[str], test_name: str, stream: bool) -> int:
+    """Executes a single test command"""
+    print_fancy(f"Running {test_name}: ", Fore.MAGENTA, flush=True, end="\n" if stream else "")
+
+    if stream:
+        result = subprocess.run(test_cmd, text=True)
+        print_fancy(f"Running {test_name}: ", Fore.MAGENTA, flush=True, end="")
+    else:
+        result = subprocess.run(test_cmd, capture_output=True, text=True)
+
+    output = (result.stdout or result.stderr or "").strip()
+
+    if result.returncode != 0 or output.endswith("FAIL"):
```

Comment:
maybe add comment that the "FAIL" is needed due to invalid return codes in unity tests?

Reply:
Referring to our previous conversation, I believe that all binaries should simply return a non-zero exit code if a failure occurs (which is already on PR in tests for the main project). This would make it sufficient to just check the return code.

Reply:
Yes, I thought that it was left intentionally until we would clean-up the tests (that's why I insisted on comment regarding WHY it was needed). I much more prefer removing it altogether :).